src/badguy/badguy.cpp

```diff
@@ -775,7 +775,7 @@ BadGuy::might_fall(int height) const
     x1 = m_col.m_bbox.get_right();
     x2 = m_col.m_bbox.get_right() + 1;
   }
-  return Sector::get().is_free_of_statics(Rectf(x1, y1, x2, y2));
+  return Sector::get().is_free_of_statics(Rectf(x1, y1, x2, y2)) && Sector::get().is_free_of_specifically_movingstatics(Rectf(x1, y1, x2, y2));
```

Comment:
The `Rectf(x1, y1, x2, y2)` rectangle could be created in a `const Rectf` variable above and then copied, saving another initialization.